src/sql/connection.py

```diff
@@ -693,14 +696,22 @@ def __init__(self, payload, engine=None, alias=None):
         if engine is None:
             raise ValueError("Engine cannot be None")
 
-        connection_name_ = "custom_driver"
+        # detect if the engine is a native duckdb connection
+        _is_duckdb_native = _check_if_duckdb_dbapi_connection(engine)
+        connection_name_ = "duckdb" if _is_duckdb_native else "custom_driver"
+
         self.url = str(engine)
         self.name = connection_name_
         self.dialect = connection_name_
-        self.session = CustomSession(self, engine)
+        self.session = DBAPISession(self, engine)
 
         self.connections[alias or connection_name_] = self
 
         self.connect_args = None
         self.alias = alias
         Connection.current = self
+
+
+def _check_if_duckdb_dbapi_connection(conn):
+    """Check if the connection is a native duckdb connection"""
+    return hasattr(conn, "df") and hasattr(conn, "pl")
```

Comment:
What is `df` and `pl`? Are these specific to DuckDB only?

Reply:
yes. they're specific to duckdb

Reply:
added a note